tests/browser/adapter-setups/adapter-setup.ts

```diff
@@ -14,15 +14,15 @@
  * limitations under the License.
  */
 
-import { AdapterInterface } from '../../../src/adapters/AdapterInterface';
+import { AdapterInterface } from '../../../src/adapters/adapter-interface';
 
 export type DoneCallback = () => void;
 
 export interface AdapterTestSetup {
   name: string;
   editorElement: string;
   inputFormat: string;
-  setEditorContent: (text: string, done: DoneCallback) => void;
+  setEditorContent: (text: string) => void | Promise<void>;
```

Comment:
I'm not sure but the done callback might be slightly better. 
So if you invoke this function you can never know whether it will return void or a promise. So you need to check and handle it accordingly if you need. However with the done callback if you want to wait for the completion or do additional thing after that, just provide the callback to it to for doing the additional logic it is simpler. With the return type  `void | Promise<void>`  you can never know which one is returning either a promise or undefined!?

Reply:
Done callback is generally deprecated. It's a old style which was used before promises. https://vitest.dev/guide/migration.html#done-callback.

We can have two different functions instead setEditorContentSync and setEditorContentAsync

Reply:
> We can have two different functions instead setEditorContentSync and setEditorContentAsync
Yeah, I think that would be better if we also need Async

Reply:
The dual nature was is unused, it has no impact. removed it